### PR TITLE
docs: flame_rive docs added

### LINKED ISSUE #1963: Bridge package docs: `flame_rive`
# Bridge package docs for `flame_rive`

This bridge package should have its own documentation section on the docs site, similarly to `flame_audio` for example:
https://docs.flame-engine.org/main/flame_audio/flame_audio.html

For inspiration (and copy-pasting) you can have a look at the documentation for the package written in the `README.md` file and the documentation that the package is bridging too.

This PR should include an interactive code example to show a `RiveComponent` live in the docs.

# Good to know

Make sure that you comment on the issue that you will be working on this before starting to program.

To set up the Flame environment you need to follow the instructions in our [contributing.md](https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md#environment-setup) doc, if you have any problems setting up your environment (or other issues of course) feel free to ask for help in the Flame channel on our [Discord server](https://discord.com/invite/pxrBmy4).

Since this is a documentation change, you need to be able to build the docs. You can do that by running `melos sphinxdoc` after you have followed the instructions in the contributing.md doc and also [installed Sphinx on your system](https://www.sphinx-doc.org/en/master/usage/installation.html). After you have built the docs you can open the `doc/_build/html/index.html` file in your web browser to see the changes.

Make sure to include a video of the example when you upload the PR so that it will be easier for us to review it. (You can attach videos directly in the PR, so no need to upload them somewhere else)

**Happy hacking!**

### PR DESCRIPTION
**Pull request by dipakp2726**
# Description
added Bridge package docs for flame_rive

## Checklist
- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have followed the [Contributor Guide] when preparing my PR.
- [ ] I have updated/added tests for ALL new/updated/fixed functionality.
- [ ] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [ ] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?
- [x] No, this PR is not a breaking change.

## Related Issues
Closes #1963

https://user-images.githubusercontent.com/36990991/197525952-ceff6b8e-e8d8-4354-b7e8-56212b35a880.mov

### CODE REVIEW

**Review by renancaraujo — APPROVED** [review]

File: doc/bridge_packages/flame_rive/rive.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# flame_rive
+
+`flame_rive` is a bridge library for using [rive](https://rive.app/) animations in your
+Flame game. Rive is a real-time interactive design and animation tool. you can use it to create
+animation
```
Comment (spydon) [inline doc/bridge_packages/flame_rive/rive.md:5]: Remember that 100 is the max line length and we should only create a newline when it overflows.
```suggestion
`flame_rive` is a bridge library for using [rive](https://rive.app/) animations in your Flame game.
Rive is a real-time interactive design and animation tool. you can use it to create animation
```
Also The last sentence is not finished, and does not start capitalized.

File: doc/bridge_packages/flame_rive/rive.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+To use it in your game you just need to add `flame_rive` to your pubspec.yaml, as can be seen in the
+[Flame Rive example](https://github.com/flame-engine/flame/tree/main/packages/flame_rive/example)
+and in the pub.dev [installation instructions](https://pub.dev/packages/flame_rive).
+
+
+## How to use
```
Comment (spydon) [inline doc/bridge_packages/flame_rive/rive.md:12]:
```suggestion
## How to use it
```

File: doc/bridge_packages/flame_rive/rive.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+First, add `animation.riv` file in the assets folder. Then load the artboard of the animation to the
+game using the `loadArtboard` method. after, create `StateMachineController` from the loaded
```
Comment (spydon) [inline doc/bridge_packages/flame_rive/rive.md:15]: Same here, remember to capitalize the sentences.
Also the sentence is not grammatically correct.

Comment (spydon) [reply doc/bridge_packages/flame_rive/rive.md:15]: Check the rest of the doc too for non-capitalized sentences, there are several more of them.

File: doc/bridge_packages/flame_rive/rive.md (hunk 1 above)
Comment (dipakp2726) [reply doc/bridge_packages/flame_rive/rive.md]: Screen Shot 2022-10-24 at 7 29 36 PM

in the editor, it shows there is no line break but when I open changes in the browser they are not the same as in the editor, the bar on the right side is at 100 char,

not sure why they look different

File: doc/bridge_packages/flame_rive/rive.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
+# flame_rive
+
+`flame_rive` is a bridge library for using [rive](https://rive.app/) animations in your Flame game.
+Rive is a real-time interactive design and animation tool. You can use it to create animation.
```
Comment (spydon) [inline doc/bridge_packages/flame_rive/rive.md:4]:
```suggestion
Rive is a real-time interactive design and animation tool and you use it to create animations.
```

**Comment by spydon** [comment]
As stated by the issue:
> This PR should include an interactive code example to show a RiveComponent live in the docs.
So don't forget to add an interactive example.

**Comment by dipakp2726** [comment]
https://user-images.githubusercontent.com/36990991/197690846-e5991650-d056-45a7-9f83-ff62fa11f914.mov

**Comment by spydon** [comment]
> Screen.Recording.2022-10-25.at.11.02.53.AM.mov

Nice! It is not interactive though? So add a tap or something to change the animation.

**Comment by dipakp2726** [comment]
https://user-images.githubusercontent.com/36990991/197985904-22d8a16a-7be3-4d21-be30-a61272c0dfd5.mov

File: doc/flame/examples/lib/rive_example.dart [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import 'dart:async';
+
+import 'package:flame/events.dart';
+import 'package:flame/game.dart';
+import 'package:flame_rive/flame_rive.dart';
+
+// ignore: depend_on_referenced_packages
```
Comment (spydon) [inline doc/flame/examples/lib/rive_example.dart:7]: You shouldn't ignore this, you should add it to the pubspec.

Comment (dipakp2726) [reply doc/flame/examples/lib/rive_example.dart]: fixed!

Comment (spydon) [reply doc/flame/examples/lib/rive_example.dart]: Now the other fix will be merged in a few minutes which will make it possible for you to remove this import completely (we're exporting rive from flame_rive).

File: doc/flame/examples/lib/rive_example.dart [hunk 6]
```diff
@@ -0,0 +1,36 @@
+import 'dart:async';
+
+import 'package:flame/events.dart';
+import 'package:flame/game.dart';
+import 'package:flame_rive/flame_rive.dart';
+
+import 'package:rive/rive.dart';
```
Comment (spydon) [inline doc/flame/examples/lib/rive_example.dart:7]:
```suggestion
```

File: doc/flame/examples/pubspec.yaml [hunk 7]
```diff
@@ -9,12 +9,16 @@ environment:
 
 dependencies:
   flame: ^1.4.0
+  flame_rive: ^1.5.2
   flutter:
     sdk: flutter
+  rive: ^0.9.1
+
```
Comment (spydon) [inline doc/flame/examples/pubspec.yaml:16]:
```suggestion
```

**Review by spydon — APPROVED** [review]

**Pull request merged**